myvpn-client/main.go

```diff
@@ -43,30 +46,59 @@ func main() {
 	defer conn.Close()
 	glog.Infoln("connected to server", conn.LocalIPAddr())
 
-	tun, err := tun.NewTUN("", conn.LocalIPAddr(), conn.IPNetMask())
+	tun, err := tun.NewTUN("")
 	if err != nil {
 		glog.Fatalln(err)
 	}
 	defer tun.Close()
 
-	err = tun.Up(upScript, conn.ExternalRemoteIPAddr().String())
+	err = tun.Run(addrScript, (&net.IPNet{conn.LocalIPAddr(), conn.IPNetMask()}).String())
 	if err != nil {
 		glog.Fatalln(err)
 	}
-	defer tun.Down(downScript, conn.ExternalRemoteIPAddr().String())
+	err = tun.Run(upScript, conn.ExternalRemoteIPAddr().String())
```

Comment:
tun.Run(...) to call varies scripts seems to be a little confusing in terminology, in my humble opinion. How about:
tun.NewTun(...) to pass in all the scripts name, so that tun knows what to call in below function:
tun.Up(conn.ExternalRemoteIPAddr().String())
tun.AddAddr((net.IPNet{conn.LocalIPAddr(), conn.IPNetMask()}).String())
tun.Down(conn.ExternalRemoteIPAddr().String())
...
etc?